### PR TITLE
change to return false when a nullable argument is passed

### PR DESCRIPTION
**Pull request by raccoonback**
I have used i18next, react-i18next.
However, when using the i18n object to determine the existence of the key, I found that passing one `nullish factor(undefined or null, or '')` results in the error below:

```js
TypeError
Cannot read property 'forEach' of undefined
```

I think `i18n.exist(undefined or null, or '')` should return` false`.

Similar to mine, there will be many cases that deliver nullish factors, and I think it is obvious that nullish factors do not exist as keys in the resource.

#### Checklist

- [ ] only relevant code is changed (make a diff before you submit the PR)
- [x] run tests `npm run test`
- [x] tests are included

#### Checklist (for documentation change)

- [ ] only relevant documentation part is changed (make a diff before you submit the PR)
- [ ] motivation/reason is provided

### CODE REVIEW

**Comment by jamuhl** [comment]
wouldn't it be cleaner to add a check in the exists function itself than in the reused resolve? Like it's already done for the translate here: https://github.com/i18next/i18next/blob/master/src/Translator.js#L87

**Comment by raccoonback** [comment]
@jamuhl 

I like your opinion, 
but wouldn't it be better to deal with common within the `resolve() function` because this error is from the `resolve() function`?

because `nullable values(undefined or null, or '')` can be passed to `keys parameter` elsewhere using the resolve() function`.

**Comment by jamuhl** [comment]
resolve is internal...currently translate, exist are "public" functions - as translate handles the return value for nullish already it seems logical for me to let exists handle it itself too

**Comment by raccoonback** [comment]
@jamuhl 
Yes, I understand. I'll change it.

**Comment by raccoonback** [comment]
@jamuhl 
I've reflected all the reviews. Please check it again.

File: src/Translator.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import baseLogger from './logger.js';
 import EventEmitter from './EventEmitter.js';
 import postProcessor from './postProcessor.js';
 import * as utils from './utils.js';
+import { isOptional } from './utils.js';
```
Comment (jamuhl) [inline src/Translator.js:5]: does not need an additional import -> can be accessed as utils.isOptional

personally I wouldn't add a helper in utils for that but have those written inline:

translate -> `if (keys === undefined || keys === null) return '';`
exists -> `if (keys === undefined || keys === null) return false;`


having a reusable helper function makes no sense -> it's not usable in other places, isOptional is non-descriptive if not knowing where it is used...

**Review by jamuhl — CHANGES_REQUESTED** [review]

File: src/Translator.js (hunk 1 above)
Comment (raccoonback) [reply src/Translator.js:5]: Yes, I'll change it to inline.

**Comment by raccoonback** [comment]
@jamuhl 
I've reflected all the your review. Please check it one more. 😃

**Comment by adrai** [comment]
lgtm

**Review by jamuhl — APPROVED** [review]
lgtm

**Pull request merged**